pvlib/irradiance.py

```diff
@@ -2186,6 +2186,94 @@ def erbs(ghi, zenith, datetime_or_doy, min_cos_zenith=0.065, max_zenith=87):
     return data
 
 
+def boland(ghi, zenith, datetime_or_doy, min_cos_zenith=0.065, max_zenith=87):
+    r"""
+    Estimate DNI and DHI from GHI using the Boland clearness index model.
+
+    The Boland model [1]_, [2]_ estimates the diffuse fraction, DF, from global
+    horizontal irradiance, GHI, through an empirical relationship between DF
+    and the ratio of GHI to extraterrestrial irradiance or clearness index, kt.
+
+    .. math::
+
+        \mathit{DF} = \frac{1}{1 + \exp\left(-5 + 8.6 k_t\right)}
+
+    where :math:`k_t` is the clearness index.
+
+    Parameters
+    ----------
+    ghi: numeric
+        Global horizontal irradiance in W/m^2.
+    zenith: numeric
+        True (not refraction-corrected) zenith angles in decimal degrees.
+    datetime_or_doy : int, float, array, pd.DatetimeIndex
```

Comment:
```suggestion
    datetime_or_doy : int, float, array, DatetimeIndex
```

Reply:
I prefer `pandas.DatetimeIndex` to use the intersphinx extension so that it's automatically hyperlinked to pandas documentation. Ditto for `numpy.ndarray`